cli/js/net.ts

```diff
@@ -163,7 +223,7 @@ export interface ListenOptions {
  *     listen({ hostname: "[2001:db8::1]", port: 80 });
  *     listen({ hostname: "golang.org", port: 80, transport: "tcp" })
  */
-export function listen(options: ListenOptions): Listener {
+export function listen(options: ListenOptions): Listener | Receiver {
```

Comment:
Maybe it would be nice to deduce return type from input:
```ts
function listen(options: ListenOptions & { transport: "tcp" }): Listener;
function listen(options: ListenOptions & { transport: "udp" }): Receiver;
function listen(options: ListenOptions): Listener | Receiver { /* ... */ }
```

Reply:
This can't work because there will be two functions with the same name

Reply:
TypeScript supports [function overloading](https://www.typescriptlang.org/docs/handbook/functions.html#overloads) so this should actually work I believe, just ensure that the first 2 function signatures do not have a body.